src/chocolatey/infrastructure.app/services/ChocolateyConfigSettingsService.cs

```diff
@@ -43,14 +46,27 @@ public void noop(ChocolateyConfiguration configuration)
             this.Log().Info("Would have made a change to the configuration.");
         }
 
-        public void source_list(ChocolateyConfiguration configuration)
+        public IEnumerable<ChocolateySource> source_list(ChocolateyConfiguration configuration,bool logResults)
         {
-            foreach (var source in configFileSettings.Sources)
+            foreach (var source in configFileSettings.Sources.AsQueryable())
             {
-                this.Log().Info(() => "{0}{1} - {2}".format_with(source.Id, source.Disabled ? " [Disabled]" : string.Empty, source.Value));
+                if (logResults) { 
+                    this.Log().Info(() => "{0}{1} - {2}".format_with(source.Id, source.Disabled ? " [Disabled]" : string.Empty, source.Value));
+                }
+                yield return new ChocolateySource {
+                    Id = source.Id,
+                    Value = source.Value,
+                    Disabled =  source.Disabled,
+                    Authenticated = string.IsNullOrWhiteSpace(source.Password)
+                };

```

Comment:
Nothing much -- I'm outputting the members as I receive them. It's a habit from writing PowerShell pipeline stuff, and it also means I don't have to actually create an IEnumerable to return. 

It's fine to cache the new objects and output them all at once if you prefer.